### PR TITLE
Combine checkbox value with var name for a full dict var name

### LINKED ISSUE #261: Add to proxy var regex - check for brackets for `i` etc. to be more specific
Add to proxy var regex - check for brackets.
^ to make an issue with.

_Originally posted by @plocket in https://github.com/plocket/docassemble-cucumber/pull/258#discussion_r650300535_

### LINKED ISSUE #270: Remove need for '/sign' in story table?
Now we have the sought var guaranteed, we can just detect the canvas and know the canvas needs to be drawn on.

### LINKED ISSUE #271: Reduce proxy matching to `xijklmn`
See `index vars` section of da documentation

### PR DESCRIPTION
**Pull request by plocket**
Which removes the need for the `checked` column.

#### Changed
- Remove the need for the `checked` story table column and prop by: 1) Moving the `value` of checkboxes into their var name column. 2) Moving the `checked` value of checkboxes into the `value` column.
- Combined checkbox encoding matching regex into one regex.

#### Fixed
- Regex testing for match to proxy var - both the regex itself being more strict and the test for a match being more predictable.

Also removed rows from a feature test using a story table. They weren't needed and were confusing to run into.

Addresses #264 and #242.

[Close #271 (proxy var matching regex duplicate), close #270 (/sign), close #261  (proxy var matching regex)]

### CODE REVIEW

File: lib/scope.js
Comment (plocket) [reply lib/scope.js:644]: I was making sure that it was fine without the usual values while at the same time not getting rid of it. That is, that its lack of useful value would not interfere with current functionality. There will be several changes to make to get rid of it completely, so I was holding off on that. If we were keeping it around, it would not be an empty string.

File: lib/scope.js
Comment (plocket) [reply lib/scope.js:788]: Re pushing into the arrays: That's not meant to be there, I'll have to clean that up.
Re return value: The return value needs to have `var_names` and needs `values`. When I clean up (before merging), I'll change the var name `just_names` to be `var_names`. It was a temporary name while I was experimenting with changing up that functionality. I'll have to look into what we should do about `values` here. It depends on what's happening in `getPageData()`.

Comment (plocket) [reply lib/scope.js:788]: Thanks for catching all this leftover code.

**Comment by plocket** [comment]
Updated, @niharikasingh

**Review by niharikasingh — APPROVED** [review]

**Pull request merged**
